docs/en/query_language/select.md

```diff
@@ -448,6 +448,19 @@ The table names can be specified instead of `<left_subquery>` and `<right_subque
 
 See the standard [SQL JOIN](https://en.wikipedia.org/wiki/Join_(SQL)) description.
 
+#### Syntax Limitations
+
+For multiple `JOIN`:
+
+- You cannot perform `SELECT * ...` if you use `SELECT` subqueries for joining. Using of `*` is available only for joining tables.
+- The `PREWHERE` clause is not available for multiple `JOIN`.
+
+For `ON`, `WHERE` and `GROUP BY` clauses:
```

Comment:
Arbitrary expressions can't be used in `ON`, `WHERE` and `GROUP BY` clauses, but you can define an expression in `SELECT` clause and then use it via alias in these clauses.

@4ertus2 btw what about `WITH` clause?